src/daft-plan/src/builder.rs

```diff
@@ -333,6 +474,8 @@ impl LogicalPlanBuilder {
     ) -> DaftResult<Self> {
         err_if_agg("hash_repartition", &partition_by)?;
 
+        let partition_by = self.substitute_getter_sugars_vec(partition_by)?;
```

Comment:
This feels really brittle - "we have to remember to call `self.substitute_getter_sugars[_vec]` on every expression input to the builder"?

I wonder if there is a better approach that will work.

Reply:
Can you just put this logic in the 
```
pub fn col<S: Into<Arc<str>>>(name: S) -> ExprRef {
    Expr::Column(name.into()).into()
}
```

Reply:
@colin-ho the issue is that determining how to convert the column names to valid getters requires the current logical plan, which does not exist when you create an `Expr::Column` but only once you pass it into one of the dataframe operations, which then call the builder. 

@jaychia I was hoping for a better approach too but layers further down than the builder rely on the expressions to be valid, so I'm unsure how to make this less brittle.

Reply:
From our conversation offline: we should do this in each of the logical ops. Create a new function that replaces the current logic in each op creation which returns a new field and expression, which does the syntactic sugar conversion